frozenlist/__init__.py

```diff
@@ -73,7 +77,43 @@ def __hash__(self):
         else:
             raise RuntimeError("Cannot hash unfrozen list.")
 
+    def __deepcopy__(self, memo: dict[int, object]):
+        obj_id = id(self)
 
+        # Return existing copy if already processed (circular reference)
+        if obj_id in memo:
+            return memo[obj_id]
+
+        # Create new instance and register immediately
+        new_list = self.__class__([])
+        memo[obj_id] = new_list
+
+        # Deep copy items
+        new_list._items[:] = [copy.deepcopy(item, memo) for item in self._items]
+
+        # Preserve frozen state
+        if self._frozen:
+            new_list.freeze()
+
+        return new_list
+
+    def __reduce__(self):
+        return (
+            _reconstruct_pyfrozenlist,
+            (self._items, self._frozen),
+        )
+
+
+def _reconstruct_pyfrozenlist(items: list[object], frozen: bool) -> "PyFrozenList":
```

Comment:
If we put the definition before this function, there would be no need to quote the annotation. Let's make it so.
```suggestion
def _reconstruct_pyfrozenlist(items: list[object], frozen: bool) -> PyFrozenList:
```